llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -559,33 +554,39 @@ bool SIInstrInfo::shouldClusterMemOps(ArrayRef<const MachineOperand *> BaseOps1,
                                       unsigned NumBytes) const {
   // If the mem ops (to be clustered) do not have the same base ptr, then they
   // should not be clustered
+  unsigned MaxMemoryClusterDWords = 8;
   if (!BaseOps1.empty() && !BaseOps2.empty()) {
     const MachineInstr &FirstLdSt = *BaseOps1.front()->getParent();
     const MachineInstr &SecondLdSt = *BaseOps2.front()->getParent();
     if (!memOpsHaveSameBasePtr(FirstLdSt, BaseOps1, SecondLdSt, BaseOps2))
       return false;
+
+    const SIMachineFunctionInfo *MFI =
+        FirstLdSt.getMF()->getInfo<SIMachineFunctionInfo>();
+    if (MFI->getMaxMemoryClusterDWords())
+      MaxMemoryClusterDWords = MFI->getMaxMemoryClusterDWords();
```

Comment:
Should just directly take MFI->getMaxMemoryClusterDWords()

Reply:
I have fixed this by adding a default limit value in SIInstrInfo.h, which could be accessible from both places.